jax/core.py

```diff
@@ -80,13 +80,10 @@ def jaxpr_as_fun(typed_jaxpr, *args):
   return eval_jaxpr(typed_jaxpr.jaxpr, typed_jaxpr.literals, (), *args)
 
 
-def new_jaxpr_eqn(*args):
-  return JaxprEqn(object(), *args)
-
-JaxprEqn = namedtuple('JaxprEqn', ['eqn_id', 'invars', 'outvars', 'primitive',
+JaxprEqn = namedtuple('JaxprEqn', ['invars', 'outvars', 'primitive',
                                    'bound_subjaxprs', 'params'])
+new_jaxpr_eqn = JaxprEqn
 
-JaxprEqn.__repr__ = JaxprEqn.__str__ = lambda eqn: str(pp_eqn(eqn))[:-1]
```

Comment:
Can we keep this?

Reply:
Yes. Actually I just realized I misread it: I thought the purpose was to remove the `eqn_id` tag. What's it do? Just clean up whitespace?

Reply:
(If so, maybe we can use `.rstrip()` instead for clarity.)

Reply:
Yes, that's all the `[:-1]` does, so `.rstrip()` SGTM.